### ISSUE TITLE
Poor compilation time

### ISSUE THREAD

**Issue by Billy-Sheppard** [rust-headless-chrome/rust-headless-chrome#420]
Hi @mdrokz!

I've been doing some investigating into the rather long compile times with headless_chrome. 

On [this branch](https://github.com/Billy-Sheppard/rust-headless-chrome-fork/tree/cratify) you can see my attempts at moving to a crate & sub-crate structure, but I have determined/narrowed the culprit to your excellent [auto_generate_cdp crate](https://github.com/mdrokz/auto_generate_cdp). 

Am I correct? Is there a way to leverage some sub-crates in this crate instead to speed up final compilation? E.g. if auto_generate_cdp had multiple crates my understanding is cargo could compile them using multiple threads.

Let me know your thoughts.

**Comment by mdrokz** [comment]
> Hi @mdrokz!
> 
> I've been doing some investigating into the rather long compile times with headless_chrome.
> 
> On [this branch](https://github.com/Billy-Sheppard/rust-headless-chrome-fork/tree/cratify) you can see my attempts at moving to a crate & sub-crate structure, but I have determined/narrowed the culprit to your excellent [auto_generate_cdp crate](https://github.com/mdrokz/auto_generate_cdp).
> 
> Am I correct (I've put the issue here instead of on there JIC I'm completely incorrect :P)? Is there a way to leverage some sub-crates in this crate instead to speed up final compilation? E.g. if auto_generate_cdp had multiple crates my understanding is cargo could compile them using multiple threads.
> 
> Let me know your thoughts.

Yeah the auto_generate_cdp contributes to the huge build time i dont know how https://github.com/mattsse/chromiumoxide handles it but they also seem to have a separate sub crate, i will try to look into it since this has been on my todolist for a long time meanwhile can you also look into it ? Thanks
